src/typing/typeloadFields.ml

```diff
@@ -181,6 +183,8 @@ let ensure_struct_init_constructor ctx c ast_fields p =
 				let v = alloc_var VGenerated cf.cf_name t p in
 				let ef = mk (TField(ethis,FInstance(c,params,cf))) cf.cf_type p in
 				let ev = mk (TLocal v) v.v_type p in
+				if not (Meta.has Meta.Optional v.v_meta) then
+						if opt then v.v_meta <- (Meta.Optional,[],null_pos) :: v.v_meta;
```

Comment:
Instead of the double `if` you can just use `&& opt` on the line above to make this look a bit nicer. Please also fix the indentation while you're at it.